### PR TITLE
rm will continue on object 404s

### PR DESCRIPTION
**Pull request by thomasmaclean**

### CODE REVIEW

File: gslib/commands/rm.py [hunk 1]
```diff
@@ -393,10 +393,15 @@ def RemoveFunc(self, name_expansion_result, thread_state=None):
...
-                            provider=exp_src_url.scheme)
+    try:
+      gsutil_api.DeleteObject(exp_src_url.bucket_name,
+                              exp_src_url.object_name,
+                              preconditions=self.preconditions,
+                              generation=exp_src_url.generation,
+                              provider=exp_src_url.scheme)
+    except NotFoundException as e:
+      # Retries during recursive object deletes can cause harmless 404s.
+      self.logger.info('Cannot find %s', exp_src_url)
```
Comment (dilipped) [inline gslib/commands/rm.py:404]: Curious about why we want to retry on 404. Can you please add a link to the issue this is trying to fix?

Comment (thomasmaclean) [reply gslib/commands/rm.py:404]: b/148433922

File: gslib/commands/rm.py [hunk 2]
```diff
@@ -393,10 +393,15 @@ def RemoveFunc(self, name_expansion_result, thread_state=None):
...
+      DecrementFailureCount()
```
Comment (dilipped) [inline gslib/commands/rm.py:405]: Why do we have call this? Is DeleteObject call going to increment the failure count?

Comment (thomasmaclean) [reply gslib/commands/rm.py:405]: `command.py` increments failure count in `PerformTask`. This score keeping would be off by one when the 404 is harmless. I'll add a slightly more verbose comment to explain why this happens.

**Pull request merged**